// @etherspot/free-bundler/dist/chunk-7XIITFLS.js
import {
  getFreeBundlerUrl,
  validateChainId
} from "./chunk-Q4MOWTYA.js";
import {
  DEFAULT_CLIENT_OPTIONS,
  DEFAULT_TRANSPORT_OPTIONS
} from "./chunk-TBNWZQEJ.js";

// src/client.ts
import { createClient, http } from "viem";
import { bundlerActions } from "viem/account-abstraction";
function createFreeBundler(options) {
  const bundlerUrl = options.bundlerUrl || getFreeBundlerUrl(options.chain.id);
  if (!bundlerUrl) {
    validateChainId(options.chain.id);
  }
  const transport = http(bundlerUrl, {
    timeout: options.transport?.timeout ?? DEFAULT_TRANSPORT_OPTIONS.timeout,
    retryCount: options.transport?.retryCount ?? DEFAULT_TRANSPORT_OPTIONS.retryCount,
    retryDelay: options.transport?.retryDelay ?? DEFAULT_TRANSPORT_OPTIONS.retryDelay
  });
  const client = createClient({
    transport,
    chain: options.chain,
    account: options.account,
    name: options.client?.name ?? DEFAULT_CLIENT_OPTIONS.name,
    pollingInterval: options.client?.pollingInterval ?? DEFAULT_CLIENT_OPTIONS.pollingInterval
  });
  return client.extend(bundlerActions);
}

export {
  createFreeBundler
};
