// @etherspot/free-bundler/dist/chunk-Q4MOWTYA.js
import {
  BUNDLER_CONFIGS,
  FREE_BUNDLER_URLS
} from "./chunk-TBNWZQEJ.js";

// src/utils.ts
function getFreeBundlerUrl(chainId) {
  return FREE_BUNDLER_URLS[chainId];
}
function getBundlerConfig(chainId) {
  return BUNDLER_CONFIGS[chainId];
}
function getSupportedChainIds() {
  return Object.keys(FREE_BUNDLER_URLS).map(Number);
}
function getAllBundlerConfigs() {
  return Object.values(BUNDLER_CONFIGS);
}
function getBundlerConfigsByNetwork(isTestnet) {
  return getAllBundlerConfigs().filter((config) => config.isTestnet === isTestnet);
}
function isChainSupported(chainId) {
  return chainId in FREE_BUNDLER_URLS;
}
function getChainName(chainId) {
  const config = getBundlerConfig(chainId);
  return config?.name ?? "Unknown Chain";
}
function validateChainId(chainId) {
  if (!isChainSupported(chainId)) {
    const supportedChains = getSupportedChainIds().join(", ");
    throw new Error(
      `Unsupported chain ID: ${chainId}. Supported chains: ${supportedChains}. Use a custom bundler URL for unsupported chains.`
    );
  }
}

export {
  getFreeBundlerUrl,
  getBundlerConfig,
  getSupportedChainIds,
  getAllBundlerConfigs,
  getBundlerConfigsByNetwork,
  isChainSupported,
  getChainName,
  validateChainId
};
