// @etherspot/free-bundler/dist/config.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/config.ts
var config_exports = {};
__export(config_exports, {
  BUNDLER_CONFIGS: () => BUNDLER_CONFIGS,
  DEFAULT_CLIENT_OPTIONS: () => DEFAULT_CLIENT_OPTIONS,
  DEFAULT_TRANSPORT_OPTIONS: () => DEFAULT_TRANSPORT_OPTIONS,
  FREE_BUNDLER_URLS: () => FREE_BUNDLER_URLS
});
module.exports = __toCommonJS(config_exports);
var FREE_BUNDLER_URLS = {
  // Ethereum Mainnet
  1: "https://bundler.etherspot.io/1",
  // Ethereum Sepolia
  11155111: "https://bundler.etherspot.io/11155111",
  // Arbitrum One
  42161: "https://bundler.etherspot.io/42161",
  // Arbitrum Sepolia
  421614: "https://bundler.etherspot.io/421614",
  // Optimism Mainnet
  10: "https://bundler.etherspot.io/10",
  // Optimism Sepolia
  11155420: "https://bundler.etherspot.io/11155420"
};
var BUNDLER_CONFIGS = {
  1: {
    chainId: 1,
    name: "Ethereum Mainnet",
    url: FREE_BUNDLER_URLS[1],
    isTestnet: false
  },
  11155111: {
    chainId: 11155111,
    name: "Ethereum Sepolia",
    url: FREE_BUNDLER_URLS[11155111],
    isTestnet: true
  },
  42161: {
    chainId: 42161,
    name: "Arbitrum One",
    url: FREE_BUNDLER_URLS[42161],
    isTestnet: false
  },
  421614: {
    chainId: 421614,
    name: "Arbitrum Sepolia",
    url: FREE_BUNDLER_URLS[421614],
    isTestnet: true
  },
  10: {
    chainId: 10,
    name: "Optimism Mainnet",
    url: FREE_BUNDLER_URLS[10],
    isTestnet: false
  },
  11155420: {
    chainId: 11155420,
    name: "Optimism Sepolia",
    url: FREE_BUNDLER_URLS[11155420],
    isTestnet: true
  }
};
var DEFAULT_TRANSPORT_OPTIONS = {
  timeout: 3e4,
  retryCount: 3,
  retryDelay: 150
};
var DEFAULT_CLIENT_OPTIONS = {
  name: "Free Bundler Client",
  pollingInterval: 4e3
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BUNDLER_CONFIGS,
  DEFAULT_CLIENT_OPTIONS,
  DEFAULT_TRANSPORT_OPTIONS,
  FREE_BUNDLER_URLS
});
