// @etherspot/free-bundler/dist/utils.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/utils.ts
var utils_exports = {};
__export(utils_exports, {
  getAllBundlerConfigs: () => getAllBundlerConfigs,
  getBundlerConfig: () => getBundlerConfig,
  getBundlerConfigsByNetwork: () => getBundlerConfigsByNetwork,
  getChainName: () => getChainName,
  getFreeBundlerUrl: () => getFreeBundlerUrl,
  getSupportedChainIds: () => getSupportedChainIds,
  isChainSupported: () => isChainSupported,
  validateChainId: () => validateChainId
});
module.exports = __toCommonJS(utils_exports);

// src/config.ts
var FREE_BUNDLER_URLS = {
  // Ethereum Mainnet
  1: "https://bundler.etherspot.io/1",
  // Ethereum Sepolia
  11155111: "https://bundler.etherspot.io/11155111",
  // Arbitrum One
  42161: "https://bundler.etherspot.io/42161",
  // Arbitrum Sepolia
  421614: "https://bundler.etherspot.io/421614",
  // Optimism Mainnet
  10: "https://bundler.etherspot.io/10",
  // Optimism Sepolia
  11155420: "https://bundler.etherspot.io/11155420"
};
var BUNDLER_CONFIGS = {
  1: {
    chainId: 1,
    name: "Ethereum Mainnet",
    url: FREE_BUNDLER_URLS[1],
    isTestnet: false
  },
  11155111: {
    chainId: 11155111,
    name: "Ethereum Sepolia",
    url: FREE_BUNDLER_URLS[11155111],
    isTestnet: true
  },
  42161: {
    chainId: 42161,
    name: "Arbitrum One",
    url: FREE_BUNDLER_URLS[42161],
    isTestnet: false
  },
  421614: {
    chainId: 421614,
    name: "Arbitrum Sepolia",
    url: FREE_BUNDLER_URLS[421614],
    isTestnet: true
  },
  10: {
    chainId: 10,
    name: "Optimism Mainnet",
    url: FREE_BUNDLER_URLS[10],
    isTestnet: false
  },
  11155420: {
    chainId: 11155420,
    name: "Optimism Sepolia",
    url: FREE_BUNDLER_URLS[11155420],
    isTestnet: true
  }
};

// src/utils.ts
function getFreeBundlerUrl(chainId) {
  return FREE_BUNDLER_URLS[chainId];
}
function getBundlerConfig(chainId) {
  return BUNDLER_CONFIGS[chainId];
}
function getSupportedChainIds() {
  return Object.keys(FREE_BUNDLER_URLS).map(Number);
}
function getAllBundlerConfigs() {
  return Object.values(BUNDLER_CONFIGS);
}
function getBundlerConfigsByNetwork(isTestnet) {
  return getAllBundlerConfigs().filter((config) => config.isTestnet === isTestnet);
}
function isChainSupported(chainId) {
  return chainId in FREE_BUNDLER_URLS;
}
function getChainName(chainId) {
  const config = getBundlerConfig(chainId);
  return config?.name ?? "Unknown Chain";
}
function validateChainId(chainId) {
  if (!isChainSupported(chainId)) {
    const supportedChains = getSupportedChainIds().join(", ");
    throw new Error(
      `Unsupported chain ID: ${chainId}. Supported chains: ${supportedChains}. Use a custom bundler URL for unsupported chains.`
    );
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getAllBundlerConfigs,
  getBundlerConfig,
  getBundlerConfigsByNetwork,
  getChainName,
  getFreeBundlerUrl,
  getSupportedChainIds,
  isChainSupported,
  validateChainId
});
